### PR TITLE
Icos image support

### PR DESCRIPTION
**Pull request by mikewurtz**
Add export/import capabilities to/from IBM Cloud Object Storage to the image manager as well as the slcli. Unit tests updated to include tests for COS path.

### CODE REVIEW

File: SoftLayer/CLI/image/import.py [hunk 1]
```diff
@@ -18,13 +18,38 @@
...
+@click.option('--root-key-id',
+              default="",
+              help="ID of the root key in Key Protect")
+@click.option('--wrapped-dek',
+              default="",
+              help="Wrapped Decryption Key provided by IBM KeyProtect")
+@click.option('--kp-id',
+              default="",
+              help="ID of the IBM Key Protect Instance")
+@click.option('--cloud-init',
```
Comment (allmightyspiff) [inline SoftLayer/CLI/image/import.py]: Click has an option `is_flag=True,` which should be used here. This way you don't need the `default=""` bit

File: SoftLayer/CLI/image/import.py [hunk 2]
```diff
@@ -18,13 +18,38 @@
 @click.option('--os-code',
               default="",
               help="The referenceCode of the operating system software"
-                   " description for the imported VHD")
+                   " description for the imported VHD, ISO, or RAW image")
+@click.option('--ibm-api-key',
```
Comment (allmightyspiff) [inline SoftLayer/CLI/image/import.py]: I'm not terribly familiar with IBM COS, but are these values obvious to someone who has used IBM COS? If not could you add some sort of info on where to get the various keys?

File: SoftLayer/CLI/image/import.py [hunk 3]
```diff
@@ -18,13 +18,38 @@
...
+              default="",
+              help="The IBM Cloud API Key with access to IBM Cloud Object "
+                   "Storage instance.")
+@click.option('--root-key-id',
+              default="",
+              help="ID of the root key in Key Protect")
+@click.option('--wrapped-dek',
+              default="",
+              help="Wrapped Decryption Key provided by IBM KeyProtect")
```
Comment (fode-ibm) [inline SoftLayer/CLI/image/import.py]: Change to, "Wrapped Data Encryption Key provided by IBM KeyProtect".

File: SoftLayer/CLI/image/import.py [hunk 4]
```diff
@@ -18,13 +18,38 @@
...
+@click.option('--kp-id',
+              default="",
+              help="ID of the IBM Key Protect Instance")
+@click.option('--cloud-init',
+              default="",
```
Comment (fode-ibm) [inline SoftLayer/CLI/image/import.py]: default value should be Boolean, which should change this option type to Boolean as well

File: SoftLayer/CLI/image/import.py [hunk 5]
```diff
@@ -18,13 +18,38 @@
...
+              help="Specifies if image is cloud init")
+@click.option('--byol',
+              default="",
```
Comment (fode-ibm) [inline SoftLayer/CLI/image/import.py]: default value should be Boolean, which should change this option type to Boolean as well.

File: SoftLayer/CLI/image/import.py [hunk 6]
```diff
@@ -18,13 +18,38 @@
...
+              help="Specifies if image is bring your own license")
+@click.option('--is-encrypted',
+              default="",
```
Comment (fode-ibm) [inline SoftLayer/CLI/image/import.py]: default value should be Boolean

File: SoftLayer/CLI/image/import.py [hunk 7]
```diff
@@ -18,13 +18,38 @@
...
+              help="ID of the root key in Key Protect")
+@click.option('--wrapped-dek',
+              default="",
+              help="Wrapped Decryption Key provided by IBM KeyProtect")
+@click.option('--kp-id',
+              default="",
+              help="ID of the IBM Key Protect Instance")
+@click.option('--cloud-init',
+              default="",
+              help="Specifies if image is cloud init")
```
Comment (fode-ibm) [inline SoftLayer/CLI/image/import.py]: The term, "cloud-init" should have hyphen, as it is the name of a specification. Change to, "Specifies whether image is enabled for cloud initialization using cloud-init".

File: SoftLayer/managers/image.py [hunk 8]
```diff
@@ -120,28 +120,68 @@ def edit(self, image_id, name=None, note=None, tag=None):
 
         return bool(name or note or tag)
 
-    def import_image_from_uri(self, name, uri, os_code=None, note=None):
+    def import_image_from_uri(self, name, uri, os_code=None, note=None,
+                              ibm_api_key=None, root_key_id=None,
+                              wrapped_dek=None, kp_id=None, cloud_init=None,
+                              byol=None, is_encrypted=None):
```
Comment (fode-ibm) [inline SoftLayer/managers/image.py]: Default values for these cloud_init, byol, and is_encrypted params should be Boolean False values, which would then match the description for them below.

File: SoftLayer/CLI/image/import.py [hunk 9]
```diff
@@ -18,13 +18,38 @@
 @click.option('--os-code',
               default="",
```
Comment (fode-ibm) [inline SoftLayer/CLI/image/import.py]: Consider changing default to None, rather than empty string.

File: SoftLayer/CLI/image/export.py [hunk 10]
```diff
@@ -12,17 +12,23 @@
 @click.command()
 @click.argument('identifier')
 @click.argument('uri')
+@click.option('--ibm-api-key',
+              default="",
```
Comment (fode-ibm) [inline SoftLayer/CLI/image/export.py]: Consider changing default to None, rather than empty string.

File: SoftLayer/CLI/image/import.py [hunk 11]
```diff
@@ -18,13 +18,38 @@
...
               help="The referenceCode of the operating system software"
-                   " description for the imported VHD")
+                   " description for the imported VHD, ISO, or RAW image")
+@click.option('--ibm-api-key',
+              default="",
```
Comment (fode-ibm) [inline SoftLayer/CLI/image/import.py]: Consider changing default to None, rather than empty string.

File: SoftLayer/CLI/image/import.py [hunk 12]
```diff
@@ -18,13 +18,38 @@
...
+              help="The IBM Cloud API Key with access to IBM Cloud Object "
+                   "Storage instance.")
+@click.option('--root-key-id',
+              default="",
```
Comment (fode-ibm) [inline SoftLayer/CLI/image/import.py]: Consider changing default to None, rather than empty string.

File: SoftLayer/CLI/image/import.py [hunk 13]
```diff
@@ -18,13 +18,38 @@
...
+              help="ID of the root key in Key Protect")
+@click.option('--wrapped-dek',
+              default="",
+              help="Wrapped Decryption Key provided by IBM KeyProtect")
+@click.option('--kp-id',
+              default="",
```
Comment (fode-ibm) [inline SoftLayer/CLI/image/import.py]: Consider changing default to None, rather than empty string.

File: SoftLayer/CLI/image/import.py [hunk 14]
```diff
@@ -18,13 +18,38 @@
...
+                   " description for the imported VHD, ISO, or RAW image")
+@click.option('--ibm-api-key',
+              default="",
+              help="The IBM Cloud API Key with access to IBM Cloud Object "
+                   "Storage instance.")
+@click.option('--root-key-id',
+              default="",
+              help="ID of the root key in Key Protect")
+@click.option('--wrapped-dek',
+              default="",
```
Comment (fode-ibm) [inline SoftLayer/CLI/image/import.py]: Consider changing default to None, rather than empty string.

File: SoftLayer/CLI/image/import.py (hunk 4 above)
Comment (mikewurtz) [reply SoftLayer/CLI/image/import.py]: The default values for these flags are going to be removed in the next patch. `is_flag=True,` removes the need to have a default value.

File: SoftLayer/managers/image.py (hunk 8 above)
Comment (mikewurtz) [reply SoftLayer/managers/image.py]: This will be included in the next patch.

File: SoftLayer/CLI/image/import.py (hunk 1 above)
Comment (mikewurtz) [reply SoftLayer/CLI/image/import.py]: Will change in the next commit.

File: SoftLayer/CLI/image/import.py (hunk 2 above)
Comment (mikewurtz) [reply SoftLayer/CLI/image/import.py]: I will add some links to our documentation on the ibm-api-key and the wrapped-dek.

File: SoftLayer/CLI/image/export.py (hunk 10 above)
Comment (mikewurtz) [reply SoftLayer/CLI/image/export.py]: Will add on next commit.

**Pull request merged**